File: app/src/main/java/com/jann_luellmann/thekenapp/ThekenappFragmentPagerAdapter.java
package com.jann_luellmann.thekenapp;

import android.content.Context;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class ThekenappFragmentPagerAdapter extends FragmentPagerAdapter {

    private ListFragment listFragment;
    private SettingsFragment settingsFragment;

    private Context context;

    public ThekenappFragmentPagerAdapter(Context context, @NonNull FragmentManager fm) {
        super(fm, FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        this.context = context;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                if(listFragment == null)
                    listFragment = new ListFragment();
                return listFragment;
            case 1:
                if(settingsFragment == null)
                    settingsFragment = new SettingsFragment();
                return settingsFragment;
            default:
                return new ListFragment();
        }
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getString(R.string.list_heading);
            case 1:
                return context.getString(R.string.settings_heading);
            default:
                return "Default";
        }
    }

    public ListFragment getListFragment() {
        return listFragment;
    }

    public SettingsFragment getSettingsFragment() {
        return settingsFragment;
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/ThekenappFragmentPagerAdapter.kt
package com.jann_luellmann.thekenapp

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.jann_luellmann.thekenapp.SettingsFragment
import com.jann_luellmann.thekenapp.R

class ThekenappFragmentPagerAdapter(private val context: Context, fm: FragmentManager) :
    FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    var listFragment: ListFragment? = null
    var settingsFragment: SettingsFragment? = null
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> listFragment!!
            1 -> settingsFragment!!
            else -> ListFragment()
        }
    }

    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> context.getString(R.string.list_heading)
            1 -> context.getString(R.string.settings_heading)
            else -> "Default"
        }
    }

    init {
        if (listFragment == null) listFragment = ListFragment()
        if (settingsFragment == null) settingsFragment = SettingsFragment()
    }
}